src/helpers.js

```diff
@@ -205,3 +205,51 @@ export function isDarkMode() {
             return window.matchMedia('(prefers-color-scheme: dark)').matches;
     }
 }
+
+/**
+ * @param {string} courseNumber A string that represents the course number of a course (eg. '122A', '121')
+ * @returns {number} This function returns a number with a decimal representation of the passed in string (eg. courseNumAsDecimal('122A') returns 122.1, courseNumAsDecimal('121') returns 121)
+ */
+export function courseNumAsDecimal(courseNumber) {
+    // I wanted to split the course detail number into letters and digits
+    const courseNumArr = courseNumber.split(/(\d+)/);
+    // Gets rid of empty strings in courseNumArr
+    const filtered = courseNumArr.filter((value) => value !== '');
+
+    // Return 0 if array is empty
+    if (filtered.length === 0) {
+        console.error(`No characters were found, returning 0, Input: ${courseNumber}`);
+        return 0;
+    }
+
+    const lastElement = filtered[filtered.length - 1].toUpperCase(); // .toUpperCase() won't affect numeric characters
+    const lastElementCharCode = lastElement.charCodeAt(0); // Just checks the first character of the last element in the array
+    // Return the last element of the filtered array as an integer if it represents an integer
+    if (47 < lastElementCharCode && lastElementCharCode < 58) {
+        return parseInt(lastElement);
+    }
+
+    // If the string does not have any numeric characters
+    if (filtered.length === 1) {
+        console.error(`The string did not have numbers, returning 0, Input: ${courseNumber}`);
+        return 0;
+    }
+
+    // This element is the second to last element of the array, supposedly a string of letters
+    const secondToLastElement = filtered[filtered.length - 2];
+    // The character codes for [A-J] or [a-j] will turn into the character codes for 0-9, respectively
+    const charCode = lastElement.toUpperCase().charCodeAt(0) - 'A'.charCodeAt(0) + '0'.charCodeAt(0);
```

Comment:
Instead of adding the character code for 0, you can just leave it as the difference between the character and A, then change the below range check to use 0 and 9 instead of their character codes.

Reply:
Suggestion completed